### DISCUSSION TITLE
Add a way to reference the record type variable when using with ```with``` keyword

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/csharplang#5880] [Language Ideas]
```C#
MyVeryVeryVeryVeryVeryLongInstanceName.MyAnotherVariable.MyYetAnotherVariable = MyVeryVeryVeryVeryVeryLongInstanceName.MyAnotherVariable.MyYetAnotherVariable with {  MyInt = MyVeryVeryVeryVeryVeryLongInstanceName.MyAnotherVariable.MyYetAnotherVariable.MyInt + 5};
```
Can be Annoying

So instead let me at least use something like : ```MyVeryVeryVeryVeryVeryLongInstanceName.MyAnotherVariable.MyYetAnotherVariable = it with {MyInt = it.MyInt+5};```

**Comment by svick** [comment]
There is a championed proposal (https://github.com/dotnet/csharplang/issues/5176) to allow compound assignment operators like `+=` inside `with`. Using that, your code would be:

```c#
MyVeryVeryVeryVeryVeryLongInstanceName.MyAnotherVariable.MyYetAnotherVariable = MyVeryVeryVeryVeryVeryLongInstanceName.MyAnotherVariable.MyYetAnotherVariable with {MyInt += 5};
```

So it would avoid one of the two repetitions of the variable.

To follow that approach further, maybe the other repetition could be avoided using something like a `with=` operator?

**Reply by Xyncgas** [reply]
Although there is cases like (essentially trying to use record class like a normal class)
```C#
MyRecord = MyRecord with { VariableA = CalculateGravity(VariableA)}
```
I am interested in the proposal, but it would be terrible, if we can't use the word it in english

**Comment by Xyncgas** [comment]
```it``` keyword can automatically refer to the smallest scope possible, so unlike ```this``` keyword refers to the current instance when you are using it with ```with``` keyword, it refers to the object ```with``` refers to.

We can introduce a another keyword called ```the```, which along ```it/its```, makes ```the NetworkPeersRecord with { Address = its Address.Hop()};``` Possible
